math/liblinear/files/patch-Makefile.diff

```diff
@@ -1,50 +1,39 @@
-*** Makefile.orig	Thu Apr 12 03:38:27 2012
---- Makefile	Sat Jun  9 21:12:44 2012
-***************
-*** 6,20 ****
-  OS = $(shell uname)
-  #LIBS = -lblas
-  
-! all: train predict
-  
-  lib: linear.o tron.o blas/blas.a
-  	if [ "$(OS)" = "Darwin" ]; then \
-! 		SHARED_LIB_FLAG="-dynamiclib -Wl,-install_name,liblinear.so.$(SHVER)"; \
-  	else \
-! 		SHARED_LIB_FLAG="-shared -Wl,-soname,liblinear.so.$(SHVER)"; \
-  	fi; \
-! 	$(CXX) $${SHARED_LIB_FLAG} linear.o tron.o blas/blas.a -o liblinear.so.$(SHVER)
-  
-  train: tron.o linear.o train.c blas/blas.a
-  	$(CXX) $(CFLAGS) -o train train.c tron.o linear.o $(LIBS)
---- 6,24 ----
-  OS = $(shell uname)
-  #LIBS = -lblas
-  
-! PREFIX ?= /usr/local
-! 
-! all: train predict lib
-  
-  lib: linear.o tron.o blas/blas.a
-  	if [ "$(OS)" = "Darwin" ]; then \
-! 		LIBEXT=".dylib"; \
-! 		SHARED_LIB_FLAG="-dynamiclib -install_name $(PREFIX)/lib/liblinear$${LIBEXT}"; \
-  	else \
-! 		LIBEXT=".so.$(SHVER)"; \
-! 		SHARED_LIB_FLAG="-shared -Wl,-soname,liblinear$${LIBEXT}"; \
-  	fi; \
-! 	$(CXX) $${SHARED_LIB_FLAG} linear.o tron.o blas/blas.a -o liblinear$${LIBEXT}
-  
-  train: tron.o linear.o train.c blas/blas.a
-  	$(CXX) $(CFLAGS) -o train train.c tron.o linear.o $(LIBS)
-***************
-*** 34,37 ****
-  clean:
-  	make -C blas clean
-  	make -C matlab clean
-! 	rm -f *~ tron.o linear.o train predict liblinear.so.$(SHVER)
---- 38,41 ----
-  clean:
-  	make -C blas clean
-  	make -C matlab clean
-! 	rm -f *~ tron.o linear.o train predict liblinear.*
+--- Makefile.orig	2023-02-17 20:01:34.000000000 +0800
++++ Makefile	2023-05-15 18:46:27.000000000 +0800
+@@ -3,18 +3,22 @@
+ CFLAGS = -Wall -Wconversion -O3 -fPIC
+ LIBS = blas/blas.a
+ #LIBS = -lblas
+-SHVER = 5
++#SHVER = 5
++PREFIX ?= /opt/local
```

Comment:
You've changed the original patch's `/usr/local` to `/opt/local` here. I guess the idea with patchfiles like these is that they should be written in such a way that upstream could apply them. We should be sending these patches to the developers so that in future we don't have to maintain them anymore. For upstream purposes, /usr/local is a better default than /opt/local. For MacPorts purposes, we override it with a build arg so it doesn't matter.

```suggestion
+PREFIX ?= /usr/local
```

Ideally the patchfile should begin with the URL of the bug report or pull request where the changes have been submitted to upstream, or the URL of the commit where they already applied it.